main.tf

```diff
@@ -77,9 +86,9 @@ resource "aws_subnet" "public" {
   vpc_id                  = aws_vpc.default.id
 
   tags = merge(
-    { "Name" = "${var.prepend_resource_type ? "subnet-" : ""}${var.name}-public-${local.az_ids[count.index]}" },
+    var.tags,
     var.public_subnet_tags,
-    var.tags
+    { "Name" = "${var.prepend_resource_type ? "subnet-" : ""}${var.name}-public-${local.az_ids[count.index]}" }
```

Comment:
Is there a reason this ordering is inconsistent with the ordering of the new tags you added? I guess it makes much more sense to keep the behaviour consistent.

Reply:
There is a reason indeed, for resources that are only present once in a VPC (such as the vpc itself, or the internet gateway), it's perfectly fine is users define their own "Name" tag as well. Hoewever on these subnets, it is better to include the AZ-ID as you'd otherwise could end up with 3 public-subnets with identical names.

So for public subnets, the order of precedence is such that the module dictates name, and will overwrite any name specified in the var.tags which is already overwritten bij var.public_subnet_tags